mantis/analysis/deskew.py

```diff
@@ -139,53 +142,47 @@ def deskew_data(
         If false, only compute the deskewed volume within a cuboid region.
     average_n_slices : int, optional
         after deskewing, averages every n slices (default = 1 applies no averaging)
-    order : int, optional
-        interpolation order (default 1 is linear interpolation)
-    cval : float, optional
-        fill value area outside of the measured volume (default None fills
-        with the minimum value of the input array)
     Returns
     -------
     deskewed_data : NDArray with ndim == 3
         axis 0 is the Z axis, normal to the coverslip
         axis 1 is the Y axis, input axis 2 in the plane of the coverslip
         axis 2 is the X axis, the scanning axis
     """
-    if cval is None:
-        cval = np.min(np.ravel(raw_data))
-
     # Prepare transforms
     Z, Y, X = raw_data.shape
 
     ct = np.cos(ls_angle_deg * np.pi / 180)
-    Z_shift = 0
-    if not keep_overhang:
-        Z_shift = int(np.floor(Y * ct * px_to_scan_ratio))
 
     matrix = np.array(
         [
             [
                 -px_to_scan_ratio * ct,
                 0,
                 px_to_scan_ratio,
-                Z_shift,
+                0,
             ],
-            [-1, 0, 0, Y - 1],
-            [0, -1, 0, X - 1],
+            [-1, 0, 0, 0],
+            [0, -1, 0, 0],
+            [0, 0, 0, 1],
         ]
     )
     output_shape, _ = get_deskewed_data_shape(
         raw_data.shape, ls_angle_deg, px_to_scan_ratio, keep_overhang
     )
 
-    # Apply transforms
-    deskewed_data = scipy.ndimage.affine_transform(
-        raw_data,
-        matrix,
-        output_shape=output_shape,
-        order=order,
-        cval=cval,
-    )
+    # to tensor on GPU
+    if torch.cuda.is_available():
+        raw_data = transforms.ToDevice("cuda")(torch.tensor(raw_data))
```

Comment:
This can be replaced by:

```suggestion
        raw_data = torch.tensor(raw_data).to("cuda")
```

However this only works for one GPU. I think it's better to explicitly pass in the GPU ID, in which case it might make sense to make this a Tensor->Tensor function and handle the device one level up.

Reply:
Thanks @ziw-liu, that's much better. 

I'm less certain about making this a Tensor->Tensor function, since this would be the first function in shrimpy that uses that pattern. We currently use `process_single_position` to call this (and other) processing functions, so we would need to make `process_single_position` capable of converting to Tensors for some specific functions.    

I think we should handle this question during our next pipelining hackathon. 

Any thoughts @ieivanov?

Reply:
Fantastic, thank you! I'll review this in detail. On the mantis computer, we do now have two GPUs, but I've set one up for display and the other for compute - the nvidia driver allows for that functionality. I have used `tensor.to('cuda')` calls and they get sent to the right GPU. I'm not so concerned for processing on the mantis computer, but we should check if anything bad happens on bruno. I do think that this function should take and return ndarrays.

Reply:
the above should work if you always have a GPU, which seems to be our case. 
If you want to make it agnostic between them:
```
device = torch.device("cuda" if torch.cuda.is_available() else "CPU")
raw_data = torch.tensor(raw_data).to(device)
```

Reply:
I think for now I'll add a `device='cpu'` argument to the deskew function such that it executed on CPU by default. This will not break our processing pipelines and as Talon pointed out, the end-to-end gain on the CLI call is smaller. The big win I see is for processing of freshly acquired data still in memory which will happen as a directly python call, possibly with `device='cuda'` argument.

Reply:
I don't think any additional change is needed for the behavior you're describing @edyoshikun @ieivanov. 

I tested the implementation in this branch on a machine without a GPU and it still runs. My impression is that if you make a `tensor` it will be on the `cpu` by default, and here I'm only transferring to the gpu if `torch.cuda.is_available()`.

Reply:
Right, but you'll always transfer it to GPU if one if available and that sometimes may not be desirable - e.g. if only a small gpu is available. I'm working on this now

Reply:
Gotcha, makes sense. Thanks.